src/app/Fake.DotNet.Cli/DotNet.fs

```diff
@@ -48,7 +48,16 @@ let mutable DefaultDotNetCliDir =
 /// ## Parameters
 ///
 /// - 'dotnetCliDir' - dotnet cli install directory
-let private dotnetCliPath dotnetCliDir = dotnetCliDir @@ (if Environment.isUnix then "dotnet" else "dotnet.exe")
+let private dotnetCliPath dotnetCliDir = 
+    let defaultCliPath = dotnetCliDir @@ (if Environment.isUnix then "dotnet" else "dotnet.exe")
+    match File.Exists defaultCliPath with
+    | true -> defaultCliPath
+    | _ -> 
+        Process.tryFindFileOnPath "dotnet"
```

Comment:
does this work on windows? I can never remember this.

The only other question is if we need to know when we take this path, because we need to ensure that this is the correct version? (or should we implicitly assume the dotnet installer script did this check?)...

Reply:
It works on Windows, that's what I use :)

When it comes to versioning, I believe dotnet CLI chooses the appropriate SDK based on [global.json](https://docs.microsoft.com/en-us/dotnet/core/tools/global-json). If there is no global.json present it will select the newest one.

While there is a risk of getting the wrong SDK-version, I would say it is ultimately the responsibility of the consumer to specify either a path or an SDK-version in global.json (that she knows is installed).

Reply:
`global.json` is highly frustrating for developers when switching between projects. So I personally don't use it. Instead I trust FAKE to ensure it uses the correct version when I specify the version explicitly... (Honestly, I'm OK with using a different version locally, as long as CI checks against a specific version)

But I'll take a look, thanks for the change!